### PR TITLE
fix issue #33

### PR DESCRIPTION
**Pull request by alexshavelev**

### CODE REVIEW

**Comment by alexshavelev** [comment]
please look for tests https://travis-ci.org/mysql-otp/mysql-otp/jobs/130820684
fails dont depend on improvements

**Comment by zuiderkwast** [comment]
Nice! Any clue why the tests fail at Travis? Is mysql is not running? I'd like to fix that before I merge this PR.

File: src/mysql.erl [hunk 1]
```diff
@@ -479,12 +479,15 @@ init(Opts) ->
...
     %% Connect socket
-    SockOpts = [{active, false}, binary, {packet, raw} | TcpOpts],
+    SockOpts = [binary, {packet, raw} | TcpOpts],
     {ok, Socket} = gen_tcp:connect(Host, Port, SockOpts),
 
     %% Exchange handshake communication.
     Result = mysql_protocol:handshake(User, Password, Database, gen_tcp,
                                       Socket),
+    inet:setopts(Socket, [{active, once}]),
+    MonitorRef = erlang:monitor(process, self()),
```
Comment (zuiderkwast) [inline src/mysql.erl]: Why monitor ourself? If we are dead, we won't get the 'DOWN' message anyway, do we?

**Comment by zuiderkwast** [comment]
The failing tests are because of the changes in this branch. You forgot some places where the connection needs to be set to passive. The first one is in mysql_protocol:handshake (on the first recv) because the connection is in active mode here. We need to connect with {active, false} like before. The other fails are also because the connection needs to be set to passive before the calls to in mysql_protocol:quit and mysql_protocol:ping and set back to active once afterwards. Please try and make the tests working.

Also please explain why we are monitoring ourself. I don't understand this.

Thanks for your effort!

**Comment by zuiderkwast** [comment]
Any news on this?

**Comment by zuiderkwast** [comment]
Closing this in favor of #41. Thanks!

**Pull request closed without merging**